ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Update.scala

```diff
@@ -197,17 +202,54 @@ object Update {
 
   /** Signal that a command submitted via [[WriteService]] was rejected.
     *
-    * See the different [[RejectionReason]] for why a command can be
-    * rejected.
+    * @param recordTime     The record time of the completion
+    * @param submitterInfo  The information provided by the submitter of the command that
+    *                       created this transaction.
+    * @param reasonTemplate A template for generating the gRPC status code with error details.
+    *                       See ``error.proto`` for the status codes of common rejection reasons.
     */
   final case class CommandRejected(
       recordTime: Timestamp,
       submitterInfo: SubmitterInfo,
-      reason: RejectionReason,
+      reasonTemplate: CommandRejected.RejectionReasonTemplate,
   ) extends Update {
     override def description: String = {
-      s"Reject command ${submitterInfo.commandId}: $reason"
+      s"Reject command ${submitterInfo.commandId}${if (cancelled) " (cancelled)"}: ${reason.message}"
     }
+
+    /** If false, the [[ReadService]]'s deduplication and submission rank guarantees
+      * apply to this rejection. The participant state implementations should
+      * strive to set this flag to false as often as possible so that applications
+      * get better guarantees.
+      */
+    lazy val cancelled: Boolean = !reason.details.exists { any =>
+        if (any.is(com.google.rpc.ErrorInfo.class)) {
+          Try(any.unpack(com.google.rpc.ErrorInfo.class)).exists( errorInfo =>
+            errorInfo.getMetadataOrDefault("definite_answer", "false") == "true"
+          )
+        } else false
+      }
```

Comment:
This doesn't work as is because we don't yet have a completed reason. Move this method into the template.